firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,544 @@
+            'No application-wide add-ons have been disabled by the update'
+
+    @property
+    def build_info(self):
+        """Return information of the current build version
+
+        :returns: A dictionary of build information
+        """
+
+        return {
+            'buildid': self.app_info.appBuildID,
+            'channel': self.update_channel.channel,
+            'disabled_addons': self.prefs.get_pref('PREF_DISABLED_ADDONS'),
+            'locale': self.app_info.locale,
+            'mar_channels': self.mar_channels.channels,
+            'url_aus': self.update_url(True),
+            'user_agent': self.app_info.user_agent,
+            'version': self.app_info.version
+        }
+
+    def force_fallback(self):
+        """Update the update.status file and set the status to 'failed:6'"""
+        with open(os.path.join(self.staging_directory, 'update.status'), 'w') as f:
+            f.write('failed: 6\n')
+
+    def get_update_at(self, update_index):
+        """Use nsIUpdateManager.getUpdateAt() to return an update
+
+        :returns: nsIUpdate object
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let update_index = arguments[0];
+            return ums.getUpdateAt(update_index);
+
+        """, script_args=[update_index])
+
+    def init_update_tests(self, fallback=False):
+        """Initialize all the data used by update tests.
+
+        :param fallback: It is a fallback update
+        """
+
+        # Prepare persisted object for update results
+        # If an update fails the post build will be the same as the pre build.
+        persisted = {
+            'update': {
+                'index': 0,
+                'staging_path': self.staging_directory},
+            # Create results object with information of the unmodified pre build
+            # TODO: Lets change to persisted.update.results once the issue is fixed:
+            # https://github.com/mozilla/mozmill-automation/issues/175
+            'updates': [
+                {
+                    'build_pre': self.build_info,
+                    'build_post': self.build_info,
+                    'fallback': fallback,
+                    'patch': {},
+                    'success': False}]}
+
+        # If requested modify the default update channel. It will be active
+        # after the next restart of the application
+        if persisted['update']['channel']:
+            # Backup the original content and the path of the channel-prefs.js file
+            persisted['update']['default_update_channel'] = {
+                'content': self.update_channel.config_file_contents,
+                'path': self.update_channel.config_file_path}
+            self.update_channel.set_default_channel(persisted['update']['channel'])
```

Comment:
Given that the UpdateChannel class lives in Python now we won't have to store the defaults anymore, but simply ensure to restore the file with its original content at the end of the test. Maybe we can call this method `setUp` and add a new `tearDown` method to do that?